basic-application/basic-application-webapp/src/main/java/fr/openwide/core/basicapp/web/application/referential/component/CityListPanel.java

```diff
@@ -0,0 +1,128 @@
+
+import fr.openwide.core.basicapp.core.business.referential.model.City;
+import fr.openwide.core.basicapp.core.business.referential.model.search.CitySort;
+import fr.openwide.core.basicapp.core.util.binding.Bindings;
+import fr.openwide.core.basicapp.web.application.referential.form.AbstractGenericListItemPopup;
+import fr.openwide.core.basicapp.web.application.referential.form.CityPopup;
+import fr.openwide.core.basicapp.web.application.referential.model.CityDataProvider;
+import fr.openwide.core.jpa.more.business.generic.model.EnabledFilter;
+import fr.openwide.core.wicket.more.markup.html.factory.AbstractParameterizedComponentFactory;
+import fr.openwide.core.wicket.more.markup.html.feedback.FeedbackUtils;
+import fr.openwide.core.wicket.more.markup.html.form.EnumDropDownSingleChoice;
+import fr.openwide.core.wicket.more.markup.html.form.LabelPlaceholderBehavior;
+import fr.openwide.core.wicket.more.markup.html.repeater.data.table.DecoratedCoreDataTablePanel.AddInPlacement;
+import fr.openwide.core.wicket.more.markup.html.repeater.data.table.builder.DataTableBuilder;
+import fr.openwide.core.wicket.more.markup.html.repeater.data.table.builder.state.IAddedCoreColumnState;
+import fr.openwide.core.wicket.more.markup.html.repeater.data.table.builder.state.IDecoratedBuildState;
+
+public class CityListPanel extends AbstractGenericListItemListPanel<City, CitySort, CityDataProvider> {
+	
+	private static final long serialVersionUID = -2165029373966420708L;
+	
+	private CityPopup popup;
+	
+	public CityListPanel(String id) {
+		this(id, new CityDataProvider(EnabledFilter.ENABLED_ONLY));
+	}
+	
+	public CityListPanel(String id, CityDataProvider dataProvider) {
+		super(id, dataProvider, dataProvider.getSortModel());
+		setOutputMarkupId(true);
+		
+		add(
+				popup
+		);
+	}
+	
+	@Override
+	protected City getNewInstance() {
+		return new City();
+	}
+
+	@Override
+	protected void createPopup() {
+		popup = new CityPopup("popup") {
+			private static final long serialVersionUID = 1L;
+			
+			@Override
+			protected void refresh(AjaxRequestTarget target) {
+				target.add(resultats);
+			}
+		};
+	}
+
+	@Override
+	protected AbstractGenericListItemPopup<City> getPopup() {
+		return popup;
+	}
+
+	@Override
+	protected IAddedCoreColumnState<City, CitySort> addColumns(DataTableBuilder<City, CitySort> builder) {
+		return builder
+				.addLabelColumn(new ResourceModel("business.listItem.label"), Bindings.city().label())
+						.withClass("text text-md");
+	}
+
+	@Override
+	protected IDecoratedBuildState<City, CitySort> addIn(IDecoratedBuildState<City, CitySort> builder) {
+		return builder
+				.addIn(AddInPlacement.HEADING_MAIN, new AbstractParameterizedComponentFactory<Component, Component>() {
+					private static final long serialVersionUID = 1L;
+					@Override
+					public Component create(String wicketId, final Component table ) {
+						return new SearchFragment(wicketId, getDataProvider(), table);
```

Comment:
I think this particular override should be pushed up to the `AbstractGenericListItemListPanel`, which would define an abstract method `Component createSearchForm(String wicketId, D dataProvider, DecoratedCoreDataTablePanel<City, CitySort> table)`, and **this** method would be overriden here.
The goal is to make the class for each `GenericListItem` as simple as possible.